packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -114,22 +122,35 @@ class SocketReference extends TypedEventEmitter<ISocketEvents> {
         assert(!SocketReference.socketIoSockets.has(key), 0x220 /* "socket key collision" */);
         SocketReference.socketIoSockets.set(key, this);
 
-        // The server always closes the socket after sending this message
-        // fully remove the socket reference now
-        socket.on("server_disconnect", (socketError: IOdspSocketError) => {
+        const disconnectHandler = (socketError: IOdspSocketError, reason: string, clientId?: string) => {
```

Comment:
It feels like this functor should not use reason (and even if it did, it should not be a string, but DisconnectReason type).
Basically, there are two arguments that supply same information - presence (or lack of) clientId is sufficient, and it makes code more correct (no need to add else branch and handle a case where reason is neither DisconnectReason.serverDisconnect nor DisconnectReason.disconnectDocument - someone can add new enum value and code should stop running / compiling, helping engineer to find out he/she is missing a case).